runtime/src/main/java/org/corfudb/runtime/LogReplicationUtils.java

```diff
@@ -1,14 +1,269 @@
+        try {
+            this.corfuStore = corfuStore;
+            corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, REPLICATION_STATUS_TABLE, ReplicationStatusKey.class,
+                    ReplicationStatusVal.class, null, TableOptions.fromProtoSchema(ReplicationStatusVal.class));
+            schedulerThread = Executors.newSingleThreadScheduledExecutor(new ThreadFactoryBuilder()
+                    .setNameFormat(LogReplicationUtils.class.getName())
+                    .build());
+        } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
+            log.error("Failed to open the replication status table", e);
+            throw new UnrecoverableCorfuError(e);
+        }
+    }
+
+    public void subscribe(@Nonnull LRMultiNamespaceListener clientListener, @Nonnull String namespace,
+                          @Nonnull String streamTag, @Nonnull List<String> tablesOfInterest) {
+        SubscriptionTask subscriptionTask = new SubscriptionTask(clientListener, namespace, streamTag,
+                tablesOfInterest, DEFAULT_BUFFER_SIZE);
+        schedulerThread.schedule(subscriptionTask, 0, TimeUnit.MILLISECONDS);
+    }
+
+    public void subscribe(@Nonnull LRMultiNamespaceListener clientListener, @Nonnull String namespace,
+                          @Nonnull String streamTag, @Nonnull List<String> tablesOfInterest, int bufferSize) {
+        SubscriptionTask subscriptionTask = new SubscriptionTask(clientListener, namespace, streamTag,
+                tablesOfInterest, bufferSize);
+        schedulerThread.schedule(subscriptionTask, 0, TimeUnit.MILLISECONDS);
+    }
+
+    private class SubscriptionTask implements Runnable {
+        private LRMultiNamespaceListener clientListener;
+        private String namespace;
+        private String streamTag;
+        private List<String> tablesOfInterest;
+        private int bufferSize;
+
+        SubscriptionTask(LRMultiNamespaceListener clientListener, String namespace, String streamTag,
+                         List<String> tablesOfInterest, int bufferSize) {
+            this.clientListener = clientListener;
+            this.namespace = namespace;
+            this.streamTag = streamTag;
+            this.tablesOfInterest = tablesOfInterest;
+            this.bufferSize = bufferSize;
+        }
+
+        @Override
+        public void run() {
+            CorfuStoreMetadata.Timestamp timestamp = getValidSubscriptionTimestamp();
+
+            if (bufferSize == DEFAULT_BUFFER_SIZE) {
+                corfuStore.getRuntime().getTableRegistry().getStreamingManager().subscribeAcrossNamespaces(clientListener,
+                        namespace, streamTag, tablesOfInterest, timestamp.getSequence());
+            } else {
+                corfuStore.getRuntime().getTableRegistry().getStreamingManager().subscribeAcrossNamespaces(clientListener,
+                        namespace, streamTag, tablesOfInterest, timestamp.getSequence(), bufferSize);
+            }
```

Comment:
I was thinking if we can expose the default_buffer_size in streamingManager. But since that class is consumed by clients, maybe not a good idea to give them access internal information.

Then will need what we have currently.
I'm good with it.